### ISSUE TITLE
First second mute with mic (great-refactor)

### ISSUE THREAD

**Issue by Rinnegatamante** [smealum/ctrulib#226]
Seems first mute second is still here with microphone. This is the code i'm using that gives that problem: http://pastebin.com/58biXMur

**Comment by Steveice10** [comment]
Try waiting a bit after micInit before starting sampling. The issue generally seems to be sampling too soon after taking it to power on the microphone.

In fact, can you check the power status right after micInit? If it's still talks, I might be able to add a wait for it in micInit

**Comment by Rinnegatamante** [comment]
Yeah i know i shouldn't free it, infact the free you see was related to an old version of the same piece of code (i forgot to comment it).

**Comment by Rinnegatamante** [comment]
Added this before StartSampling and nothing changed:

_do { MICU_GetPower(&power); }while(!power);

**Comment by Steveice10** [comment]
Can you try a normal sleep for a bit then? I know for sure that it works fine with the MIC example, which doesn't immediately start sampling after micInit.

**Comment by Rinnegatamante** [comment]
Using svcSleepThread(1000000000); seems to work but now i (obviously) have a sleep for a second. What could be an optimal value?

**Comment by Rinnegatamante** [comment]
Bump
